satpy/cf/area.py

```diff
@@ -0,0 +1,78 @@
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""CF processing of pyresample area information."""
+import logging
+
+import xarray as xr
+from packaging.version import Version
+from pyresample.geometry import AreaDefinition, SwathDefinition
+
+logger = logging.getLogger(__name__)
+
+
+def _add_lonlat_coords(dataarray):
+    """Add 'longitude' and 'latitude' coordinates to DataArray."""
+    dataarray = dataarray.copy()
+    area = dataarray.attrs["area"]
+    ignore_dims = {dim: 0 for dim in dataarray.dims if dim not in ["x", "y"]}
+    chunks = getattr(dataarray.isel(**ignore_dims), "chunks", None)
+    lons, lats = area.get_lonlats(chunks=chunks)
+    dataarray["longitude"] = xr.DataArray(lons, dims=["y", "x"],
+                                          attrs={"name": "longitude",
+                                                 "standard_name": "longitude",
+                                                 "units": "degrees_east"},
+                                          name="longitude")
+    dataarray["latitude"] = xr.DataArray(lats, dims=["y", "x"],
+                                         attrs={"name": "latitude",
+                                                "standard_name": "latitude",
+                                                "units": "degrees_north"},
+                                         name="latitude")
+    return dataarray
+
+
+def _create_grid_mapping(area):
+    """Create the grid mapping instance for `area`."""
+    import pyproj
+
+    if Version(pyproj.__version__) < Version("2.4.1"):
+        # technically 2.2, but important bug fixes in 2.4.1
+        raise ImportError("'cf' writer requires pyproj 2.4.1 or greater")
+    # let pyproj do the heavily lifting (pyproj 2.0+ required)
+    grid_mapping = area.crs.to_cf()
+    return area.area_id, grid_mapping
+
+
+def _add_grid_mapping(dataarray):
+    """Convert an area to at CF grid mapping."""
+    dataarray = dataarray.copy()
+    area = dataarray.attrs["area"]
+    gmapping_var_name, attrs = _create_grid_mapping(area)
+    dataarray.attrs["grid_mapping"] = gmapping_var_name
+    return dataarray, xr.DataArray(0, attrs=attrs, name=gmapping_var_name)
+
+
+def area2cf(dataarray, include_lonlats=False, got_lonlats=False):
+    """Convert an area to at CF grid mapping or lon and lats."""
+    res = []
+    if not got_lonlats and (isinstance(dataarray.attrs["area"], SwathDefinition) or include_lonlats):
```

Comment:
CodeScene is complaining about this. I suppose to make it happy you could break the conditions into their own variables. Or do something like:

```
include_lonlats = include_lonlats or isinstance(dataarray.attrs["area"], SwathDefinition)
if not got_lonlats and include_lonlats:
    ...
```